updater/lib/dependabot/dependency_change_builder.rb

```diff
@@ -145,9 +145,17 @@ def generate_dependency_files
 
     sig { params(dependencies: T::Array[Dependabot::Dependency]).returns(Dependabot::FileUpdaters::Base) }
     def file_updater_for(dependencies)
+      # Initialize filtered_files with all dependency files by default
+      filtered_files = dependency_files
+
+      # Filter files only if all dependencies use the "pub" package manager
+      if dependencies.all? { |dep| dep.package_manager == "pub" }
```

Comment:
I can't think of a reason we want support files changed for any ecosystem, so I'd suggest dropping the package_manager check. 

Also if you could write a test that would be great.

I'd also be curious why we're modifying the flutter installation for the pub ecosystem. If we knew why and could stop it that would make updates faster. But this change at the updater level seems like a good idea in general.